### DISCUSSION TITLE
2017-18 Private Schools Near Me

### DISCUSSION THREAD

**Discussion by zhuxiaoyu1019** [jakobzhao/geog495#47] [Final Project]
Application: https://zhuxiaoyu1019.github.io/geog495-final-project/
Repository: https://github.com/zhuxiaoyu1019/geog495-final-project.git
Description: Map of Private Schools (independent schools) throughout the United States in 2017-18. The primary goal of this project is to develop a private school search engine to provide basic information about private schools when the client submits a query. Searches can be performed on the top right with the geocoder or by selecting any private school listed on the left and the application will mark the location and highlight the nearest school. Due to the data size, the current application takes a longer runtime and which results in a longer waiting time to load. (The application is based on a large dataset, it might take a longer time to run and load.)
<img width="1440" alt="app-screenshot" src="https://user-images.githubusercontent.com/72243528/145249253-349b7006-f524-4629-9c11-0eaa5a2f6890.png">

**Comment by jakobzhao** [comment]
Thank you for creating this map! It could be very useful for choosing schools over the country. Couple of suggestions for your future improvements.

1) figure-ground issue: The marker for the school can be hardly seen because the color of the marker and the background are similar and the marker is mostly transparent. 
2) is there a specific reason why the basemap is black?
3) You might want to optimize the loaded geojson. for example, the digits of the decimals can be reduced from 15 to 6. And deleting unnecessary attributes.
4) the application has used several colors, like dark gray, white and green, light gray, try to use less color hue.

**Comment by VaneRaklan** [comment]
Between the map, the side panel and the tooltip, I think you did a good job with keeping the application simple and clean! Like if the data volume bottlenecking the runtime wasn't an issue, I feel like the application would be smooth like butter!
That being said, I think an interesting fix (although probably a big undertaking) could be letting users select a state to search within from a dropdown or something. You could zoom the map to the state they choose and when they search using the geocoder, the application only has to deal with schools from that state. Something along those line, again I'm not even sure of how one would go about that but I bet there's a way.

**Comment by kevography** [comment]
From a visual standpoint I like the basemap and it seems like you have a specific idea for the user interface. I have two main points that I think could be helpful:

1. I think the data on the backend is really reducing how responsive the map actually is to control. My advice here would be to find a way to slim down the file size or narrow the scope of the map so that it doesn't slow down the application.
2. Find a way to make the icons more readable on the basemap. I can understand if you made and calibrated the icons to a default basemap before you added your own. If you want to keep the current basemap (which visually, I do like) I'd recommend making the icons on the lighter side of the spectrum.

**Comment by notzulkifli** [comment]
Private school idea is an amazing one. Theres only a couple of things i could suggest in this case. The dataset as you said was huge so it makes the runtime slower. Possibly focusing on a certain section of the map could be a great way of aggregating the information and improving runtime speed, For example, "West coast private schools" "East coast private schools" "Midwest private schools" so on and so fouth. that way when we sort the data is wouldn't be on the slower side. The second thing i can think of is the icon color and map color. Maybe choosing another icon color or specificially a different icon image would make this more unique and thus standout from the map.

**Comment by manhongg** [comment]
I liked that you took a unique approach to mapping your data on the locations and accessibility of private schools in various locations. It was very interesting to read through the data points in your web app and I think you did a good job of organizing and displaying the information.I agree with some of the other feedback that it may be more effective to filter out some of the data beforehand in order to improve the loading times. Changing the design of the map so that the icons stand out more from the basemap would also be helpful in improving the readability of your web app. overall, great job with your project!

**Comment by zeyany2** [comment]
Hi! I really like this map! Just like professor said, it could be very useful for choosing schools over the country. The idea is great, and I really like how the search bar works. At the same time, I have a suggestion. You can change the base map or markers' color in order to see better. When I first look you your map, all markers are hard to see in the map.

**Comment by Subarashiihibi** [comment]
I think your locator is well designed in its appearances. But it takes really a long time to load in all of the schools from your geojson and my computer might be stuck at some point. I think maybe this is cause by your super large dataset and my suggestion is that you could filter out some states or even only specify on certain states which could definitely reduce the time spent to load all the data.
